plugins/qiskit_executor/routes.py

```diff
@@ -433,13 +405,12 @@ def render(self, db_id, data: Mapping, errors: dict, valid: bool):
             msg = f"Could not load task data with id {db_id} to read parameters!"
             raise KeyError(msg)
 
-        params: CircuitParameters = CircuitParameterSchema().loads(
-            db_task.data["parameters"]
-        )
         backend_parameter_schema = BackendParameterSchema()
-        backend_parameter_schema.fields["backend"].metadata["datalist"] = [
-            backend.name() for backend in get_backends(params.ibmqToken)
-        ]
+        backend_parameter_schema.fields["backend"].metadata[
+            "datalist"
+        ] = PluginState.get_value(
```

Comment:
I have missed that in previous reviews. Does this update the global (i.e. class level) definition of the field? if not, then this can stay, if yes, then we should pass this additional datalist info seperate from the schema to the template.

Reply:
This uses an instance of the `BackendParameterSchema`. It shouldn't affect the global definition.

https://github.com/UST-QuAntiL/qhana-plugin-runner/blob/ebf6fc369499720dcfb3f0a715d72904673e410a/plugins/qiskit_executor/routes.py#L408-L409

Reply:
This is only true if the instance level field is a deep copy of the class level backend field. If not you are just accessing and mutating the same field object through the instance.

Reply:
You were right, the global definition was updated. Now, the `values` are used to pass the backend names to the template.

https://github.com/UST-QuAntiL/qhana-plugin-runner/blob/545f5296b581a46b70df6881c1889b2153bb5025/plugins/qiskit_executor/routes.py#L394-L410

Reply:
## Update

The current implementation for `input_type == 'text_with_datalist'` checks if the parameter schema has an attribute `datalists` of type `mapping` and then creates the datalist html tag.

https://github.com/UST-QuAntiL/qhana-plugin-runner/blob/75d2ac40d29a44a1c91ca3964cb05e680db1ac4a/qhana_plugin_runner/templates/forms.html#L33-L43

https://github.com/UST-QuAntiL/qhana-plugin-runner/blob/75d2ac40d29a44a1c91ca3964cb05e680db1ac4a/plugins/qiskit_executor/routes.py#L394-L396

Reply:
is it possible to check for both, the global datalists property and a datalist in the field metadata (with priority to the global as override)? You could introduce a new helper function to keep the template looking nice (see <https://github.com/UST-QuAntiL/qhana-plugin-runner/blob/main/qhana_plugin_runner/util/jinja_helpers.py>)

Reply:
Yes. Now, it is checked if the field has a `datalist` in its metadata. If that's unsuccessful, it is checked whether the schema has a `datalists` dict with a corresponding key.